### DISCUSSION TITLE
Need help with cheats!

### DISCUSSION THREAD

**Discussion by toolatexo** [rafaelvcaetano/melonDS-android#1304] [General]
Specifically for Pokemon Platinum, I'd like to be able to cheat to get the mythics etc, but I cannot seem to figure it out. Not much to find online, did see some posts made by rafaelvcaetano himself but none of it worked, yet.

**Comment by rafaelvcaetano** [comment]
So, here is the most straightforward explanation. First, you need to download a cheat database. The most common one can be found here: https://mega.nz/folder/UNwxET6K#s__L5mln48N5gUJsWwsMZQ/folder/NEAFwLgJ
In that link, download the "cheats.xml" file. Then, in melonDS, go to Settings > Cheats, click "Import cheats" and select the downloaded file. When it's finished, make sure the "Enable cheats" options is enabled. After that, you are good to go. While playing a game, if you go to the pause menu, you will find the "Cheats" option, in which you can toggle cheats as you want.

**Reply by marczeman** [reply]
My app just crashes and exits when I try to import the XML. Using S23 Ultra

**Reply by rafaelvcaetano** [reply]
You must have Android 14 and the current release of melonDS has an issue when importing cheats on that version. If you don't want to wait for the new release, you can use the nightly releases which already have this issue fixed. You can find them here: https://github.com/rafaelvcaetano/melonDS-android/releases/tag/nightly-release

**Reply by Jellyfishblue** [reply]
How about we cheat on DSiware? Or there's no cheat for melonDS DSiware ? I already import the xml. Thank you 🙏.

**Reply by Toolsel** [reply]
> So, here is the most straightforward explanation. First, you need to download a cheat database. The most common one can be found here: https://mega.nz/folder/UNwxET6K#s__L5mln48N5gUJsWwsMZQ/folder/NEAFwLgJ In that link, download the "cheats.xml" file. Then, in melonDS, go to Settings > Cheats, click "Import cheats" and select the downloaded file. When it's finished, make sure the "Enable cheats" options is enabled. After that, you are good to go. While playing a game, if you go to the pause menu, you will find the "Cheats" option, in which you can toggle cheats as you want.

I activated the cheats but it tells me when I want to have the event pokemon and I click the check it tells me *you haven't activated any cheats in this rom*

**Reply by Toolsel** [reply]
Help pls

**Reply by bakara923** [reply]
Rafaelvaetano the real GOAT i know this is a years before but thnaks alot ts help me alot

**Comment by Hallow-33** [comment]
I tried it on hard gold but i don't know if it worked or not

**Reply by ndivinity** [reply]
I tried as well, both importing the apparently common Mega link with the cheats, imported the XML but nothing appears in the cheats menu. I tried with Pokémon HeartGold in Spanish.

And surprisingly trying with spare cheats I found googling around don't work either. Apparently the trigger with the spare cheats is select+up, but nothing, that may be a issue with the cheats although I tried from various places.

**Comment by Mickel547** [comment]
If you need to catch up with a cheat, text on HACKSOLUTION2002@ GMAIL.COM

**Reply by asimpkins93-max** [reply]
I have Pokemon SoulSilver on Android using melonds and I'm not very good with cheats can some please guide me through what I need to do? It says I need to be import cheats using xml data base but I'm not sure how to do that. please help!!!!

**Reply by FelizNavidad-D** [reply]
> I have Pokemon SoulSilver on Android using melonds and I'm not very good with cheats can some please guide me through what I need to do? It says I need to be import cheats using xml data base but I'm not sure how to do that. please help!!!!

I'm also trying to edit the xml file to add a ROM hack but I can't seem to get the correct game ID. I went online to use a hex editor to try to find the CRC32 because that's what I thought the numbers in the game ID were but then melon ds wouldn't recognize it. Then I read that cheat devices get the CRC from the first 512 bytes so I tried getting it that way and still nothing. Am I supposed to reverse the number I get or something? I think in programming it's called "little endian"??

**Reply by rafaelvcaetano** [reply]
> I'm also trying to edit the xml file to add a ROM hack but I can't seem to get the correct game ID. I went online to use a hex editor to try to find the CRC32 because that's what I thought the numbers in the game ID were but then melon ds wouldn't recognize it. Then I read that cheat devices get the CRC from the first 512 bytes so I tried getting it that way and still nothing. Am I supposed to reverse the number I get or something? I think in programming it's called "little endian"??

It is the CRC32 but only of the ROM header, that is, the first 512 bytes.

**Comment by Mickel547** [comment]
Where are you located??

**Comment by Mickel547** [comment]
Where are you located?

**Reply by asimpkins93-max** [reply]
I'm located in the US but I actually got it figured out.. I just followed the steps in a previous comment

**Comment by Mickel547** [comment]
Awesome, good luck!.

I'm working on a project, would you like to work with me pls.

It's really huge and technical, I will like to work it out with someone
